src/System Application/App/VS Code Integration/src/VsCodeIntegrationImpl.Codeunit.al

```diff
@@ -86,22 +130,59 @@ codeunit 8333 "VS Code Integration Impl."
     end;
 
     [Scope('OnPrem')]
-    local procedure FormatDependency(var NavAppInstalledApp: Record "NAV App Installed App"): Text
+    local procedure FormatDependencyAsParameter(var NavAppInstalledApp: Record "NAV App Installed App"): Text
     var
         AppVersion: Text;
-        AppVersionLbl: Label '%1.%2.%3.%4', Comment = '%1 = major, %2 = minor, %3 = build, %4 = revision', Locked = true;
         DependencyFormatLbl: Label '%1,%2,%3,%4;', Comment = '%1 = Id, %2 = Name, %3 = Publisher, %4 = Version', Locked = true;
     begin
         // Skip System and Base app
         case NavAppInstalledApp."App ID" of
             SystemApplicationIdTxt, BaseApplicationIdTxt, ApplicationIdTxt:
                 exit('')
             else
-                AppVersion := StrSubstNo(AppVersionLbl, NavAppInstalledApp."Version Major", NavAppInstalledApp."Version Minor", NavAppInstalledApp."Version Build", NavAppInstalledApp."Version Revision");
+                AppVersion := FormatDependencyVersion(NavAppInstalledApp."Version Major", NavAppInstalledApp."Version Minor", NavAppInstalledApp."Version Build", NavAppInstalledApp."Version Revision");
                 exit(StrSubstNo(DependencyFormatLbl, Format(NavAppInstalledApp."App ID", 0, 4), NavAppInstalledApp.Name, NavAppInstalledApp.Publisher, AppVersion));
         end;
     end;
 
+    [Scope('OnPrem')]
+    procedure GetDependenciesAsSerializedJsonArray(var PublishedApplication: Record "Published Application"): Text
+    var
+        Json: Codeunit Json;
+    begin
+        Json.InitializeCollection('');
+        if PublishedApplication.FindSet() then
+            repeat
+                Json.AddJObjectToCollection(FormatDependencyAsSerializedJsonObject(PublishedApplication));
+            until PublishedApplication.Next() = 0;
+
+        exit(Json.GetCollectionAsText(true));
+    end;
+
+    [Scope('OnPrem')]
+    local procedure FormatDependencyAsSerializedJsonObject(var PublishedApplication: Record "Published Application") Value: Text
+    var
+        AppVersion: Text;
+        Dependency: JsonObject;
+    begin
+        Dependency.Add('id', PublishedApplication.ID);
+        Dependency.Add('name', PublishedApplication.Name);
+        Dependency.Add('publisher', PublishedApplication.Publisher);
+        AppVersion := FormatDependencyVersion(PublishedApplication."Version Major", PublishedApplication."Version Minor", PublishedApplication."Version Build", PublishedApplication."Version Revision");
+        Dependency.Add('version', AppVersion);
+
+        Dependency.WriteTo(Value);
+        exit(Value);
```

Comment:
I would remove it, because the return value is named and was already set.
```suggestion
```

Reply:
Thanks for the review!